### ISSUE TITLE
Folders tree view unintuitive

### ISSUE THREAD

**Issue by Beep6581** [Beep6581/RawTherapee#3845]
I realized that our tree view in the Folders panel in the File Browser is actually quite unintuitive. I never noticed it myself as I'm used to using it, but seeing a new user baffled by it made me realize there is a problem there.

The tree view has two elements: a triangle ("expander arrow") and a label. Single clicking on a label does nothing. Double-clicking opens it but does not expand the tree branch. Single clicking on the expander arrow might expand if it contains a subfolder - there is no indication as to which folder contains subfolders, see prematurely-closed issue #1374. Users are expected to figure this out, but this is not intuitive.

Several things could be improved:
* Let single-clicking a cell text expand the branch if it can be expanded, the same way as single-clicking a cell pixbuf expander arrow does.
* Let the currently opened folder's node in the tree be in bold or a different (background-)color, so even if I scroll down or click elsewhere, I can still see which folder is actually opened.
* Only show an expander arrow for a folder which contain subfolders. No subfolder, no expander arrow.

**Comment by sguyader** [comment]
I agree! Even if I'm quite used to the way it works now, I still think it's awkward. The suggested changes would really improve the brosing experience.

**Comment by Beep6581** [comment]
Evidence https://discuss.pixls.us/t/rawtherapee-cannot-open-files-on-windows-10-x64/4185

**Comment by Entropy512** [comment]
This got linked to from @sguyader posting on pixls - A question, what is the rationale for not making a single click on cell text open the folder?  I'm guessing possible performance issues from switching too often?

**Comment by Beep6581** [comment]
@Entropy512 I think it should expand, at least on mouse-click.

**Comment by Beep6581** [comment]
Related: #3845 #4193 #4924
